ruby-monday-blog/app/controllers/posts_controller.rb

```diff
@@ -25,7 +26,18 @@ def show
     @tags = @post.tags
   end
 
+  protected
+
+  def authenticate_user!
+    if user_signed_in?
+      super
+    else
+      redirect_to posts_path, notice: 'You need to log in to create a post.'
```

Comment:
Specifying the kick--back path. Otherwise just calls super. There might be a better hook or method to override, not sure